packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -59,7 +59,7 @@ function hydrateText(vnode: VText, node: Node) {
         // eslint-disable-next-line lwc-internal/no-global-node
         validateNodeType(vnode, node, Node.TEXT_NODE);
 
-        if (node.nodeValue !== vnode.text) {
+        if (node.nodeValue !== vnode.text && !(node.nodeValue === '\u200D' && vnode.text === '')) {
```

Comment:
when is the nodeValue `'\u200D'` and the node text different from ''?

Reply:
when there's a mismatch: the ssr text is `''` and the client is different than `''`